src/Widgets/DeviceManagerWidget.vala

```diff
@@ -52,6 +54,29 @@ public class Sound.Widgets.DeviceManagerWidget : Gtk.Grid {
         update_showable ();
     }
 
+    private void disconnected () {
+        if (null == device_list) {
+            return;
+        }
+        foreach (unowned var child in device_list.get_children ()) {
+            device_list.remove (child);
+        }
+    }
+
+    private void update_device (Device device) {
+        if (device.input != is_input_manager) {
+            return;
+        }
+        if (null != device_list) {
```

Comment:
Can the device list itself be null? I see that it's made on the constructor and only it's childs are removed but not the list itself. So maybe the check is not required

Reply:
After fixing some of the objects not correctly disposing, this check became unnecessary. It's been removed.